src/rcl_lifecycle_bindings.cpp

```diff
@@ -0,0 +1,363 @@
+    service_name = "~/get_available_states";
+    service = &state_machine->com_interface.srv_get_available_states;
+  } else if (lifecycle_srv_field_name.compare(
+                 "srv_get_available_transitions") == 0) {
+    service_name = "~/get_available_transitions";
+    service = &state_machine->com_interface.srv_get_available_transitions;
+  } else if (lifecycle_srv_field_name.compare("srv_change_state") == 0) {
+    service_name = "~/change_state";
+    service = &state_machine->com_interface.srv_change_state;
+  }
+
+  // build result object {name: <srv_name>, handle: <rcl handle of service_t>}
+  v8::Local<v8::Object> named_srv_obj = Nan::New<v8::Object>();
+  Nan::Set(named_srv_obj, Nan::New("name").ToLocalChecked(),
+           Nan::New(service_name).ToLocalChecked());
+
+  // Note: lifecycle Services are created and managed by their
+  // rcl_lifecycle_state_machine. Thus we must not manually
+  // free the lifecycle_state_machine's service pointers.
+  // To accomplish this we create srv_handle instances with
+  // the free_ptr parameter of false. Failing to do this results
+  // in a double free() error.
+  auto srv_handle = RclHandle::NewInstance(service,
+                                          nullptr,
+                                          [] { return RCL_RET_OK; },
+                                          false);
+  Nan::Set(named_srv_obj, Nan::New("handle").ToLocalChecked(), srv_handle);
+
+  info.GetReturnValue().Set(named_srv_obj);
+}
+
+// return null if transition exists from current state
+NAN_METHOD(TriggerLifecycleTransitionById) {
+  v8::Local<v8::Context> currentContent = Nan::GetCurrentContext();
+  RclHandle* state_machine_handle = RclHandle::Unwrap<RclHandle>(
+      Nan::To<v8::Object>(info[0]).ToLocalChecked());
+  rcl_lifecycle_state_machine_t* state_machine =
+      reinterpret_cast<rcl_lifecycle_state_machine_t*>(
+          state_machine_handle->ptr());
+  int transition_id = Nan::To<int64_t>(info[1]).FromJust();
+
+  bool publish = true;
+  rcl_ret_t ret = rcl_lifecycle_trigger_transition_by_id(
+      state_machine, transition_id, publish);
+
+  if (ret == RCL_RET_OK) {
+    const rcl_lifecycle_state_t* current_state = state_machine->current_state;
+    info.GetReturnValue().Set(wrapState(current_state));
+  }
+}
+
+// return null if transition exists from current state
+NAN_METHOD(TriggerLifecycleTransitionByLabel) {
+  v8::Local<v8::Context> currentContent = Nan::GetCurrentContext();
+  RclHandle* state_machine_handle = RclHandle::Unwrap<RclHandle>(
+      Nan::To<v8::Object>(info[0]).ToLocalChecked());
+  rcl_lifecycle_state_machine_t* state_machine =
+      reinterpret_cast<rcl_lifecycle_state_machine_t*>(
+          state_machine_handle->ptr());
+  std::string transition_label(*Nan::Utf8String(info[1]));
+
+  bool publish = true;
+  rcl_ret_t ret = rcl_lifecycle_trigger_transition_by_label(
+      state_machine, transition_label.c_str(), publish);
+
+  if (ret == RCL_RET_OK) {
```

Comment:
throw on error?

Reply:
+1. Using this macro now throughout the file.
```
  THROW_ERROR_IF_NOT_EQUAL(RCL_RET_OK,
                           rcl_lifecycle_trigger_transition_by_id(
                              state_machine, transition_id, publish),
                           rcl_get_error_string().str);
```